launcher/minecraft/MinecraftInstance.cpp

```diff
@@ -1024,6 +1018,29 @@ QMap<QString, QString> MinecraftInstance::createCensorFilterFromSession(AuthSess
     return filter;
 }
 
+QMap<QString, QString> MinecraftInstance::makeProfileVarMapping(std::shared_ptr<LaunchProfile> profile) const
+{
+    QMap<QString, QString> result;
+
+    result["profile_name"] = name();
+    result["version_name"] = profile->getMinecraftVersion();
+    result["version_type"] = profile->getMinecraftVersionType();
+
+    QString absRootDir = QDir(gameRoot()).absolutePath();
+    result["game_directory"] = absRootDir;
+    QString absAssetsDir = QDir("assets/").absolutePath();
+    auto assets = profile->getMinecraftAssets();
+    result["game_assets"] = AssetsUtils::getAssetsDir(assets->id, resourcesDir()).absolutePath();
+
+    // 1.7.3+ assets tokens
+    result["assets_root"] = absAssetsDir;
+    result["assets_index_name"] = assets->id;
+
+    result["library_directory"] = QDir("libraries/").absolutePath();
```

Comment:
I know it may be more complicated, but shouldn't we use `APPLICATION->metacache()->getBasePath("libraries")`?

Reply:
Well, on download, we use the makeCached function from the net folder, and that creates a meta entity for that file using the given base.
Given that the proposed solution may be a bit too much, as from what I see now, the assets behave similarly, but we still used the ` QDir("").absolutePath()`